### PR TITLE
fwupd: 1.8.4 -> 1.8.6

### PR DESCRIPTION
**Pull request by amaxine**
- https://github.com/fwupd/fwupd/releases/tag/1.8.5

###### Description of changes

I haven't had the chance to fully test it, just compile and check fwupd.tests. There's quite extensive meson changes that led to dropping some helper scripts, so something might have broken. I haven't fully reviewed the diff.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

File: pkgs/os-specific/linux/firmware/fwupd/default.nix [hunk 1]
```diff
@@ -58,6 +58,7 @@ let
   python = python3.withPackages (p: with p; [
     pygobject3
     setuptools
+    markdown
```
Comment (jtojnar) [inline pkgs/os-specific/linux/firmware/fwupd/default.nix:61]: We should probably patch out the markdown check like we did previously.

**Comment by jtojnar** [comment]
Otherwise the diff looks good.

File: pkgs/os-specific/linux/firmware/fwupd/default.nix (hunk 1 above)
Comment (maxeaubrey) [reply pkgs/os-specific/linux/firmware/fwupd/default.nix:61]: Oh, good point, I should have read the comments I deleted :woman_facepalming:

**Comment by maxeaubrey** [comment]
@ofborg build fwupd.tests

File: pkgs/os-specific/linux/firmware/fwupd/drop-markdown-check.patch [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+--- a/meson.build
++++ b/meson.build
+@@ -539,18 +539,12 @@ fwupd_gir = []
+ gir_dep = dependency('gobject-introspection-1.0', required: get_option('introspection'))
+ introspection = get_option('introspection').disable_auto_if(host_machine.system() != 'linux').disable_auto_if(not gir_dep.found())
+ 
+-markdown_version = run_command(
+-  python3, '-c', 'import markdown; print(markdown.__version__)'
+-).stdout().strip()
+-docs_python_deps = get_option('docs').require(markdown_version.version_compare('>=3.2'),
```
Comment (jtojnar) [inline pkgs/os-specific/linux/firmware/fwupd/drop-markdown-check.patch:22]: Maybe 

```suggestion
-markdown_version = run_command(
-  python3, '-c', 'import markdown; print(markdown.__version__)'
-).stdout().strip()
-docs_python_deps = get_option('docs').require(markdown_version.version_compare('>=3.2'),
+docs_python_deps = get_option('docs').require(true,
```

will be simpler.

Comment (maxeaubrey) [reply pkgs/os-specific/linux/firmware/fwupd/drop-markdown-check.patch:22]: Fair, that'll be easier to rebase in the future. I'll fix it up later when I'm near a computer again.

**Comment by michaeladler** [comment]
The current fwupd on nixos-unstable seems to be broken:

```
# fwupdmgr get-remotes
No remotes configured
```

Seems to be related to https://github.com/fwupd/fwupd/issues/1555 although we don't have ConfigurationDirectory set in the systemd unit. Does it work with your 1.8.5 build? 
Btw, meanwhile 1.8.6 has been released.

**Comment by maxeaubrey** [comment]
I'm closing this, at a quick glance 1.8.6 requires more work to rebase patches. Unfortunately I don't have the time to look at this in enough depth right now.

**Comment by jtojnar** [comment]
Maybe 1.8.5 is enough, it supposedly fixes a CVE https://repology.org/project/fwupd/cves?version=1.8.4

**Comment by maxeaubrey** [comment]
Agreed, that doesn't look like it has any potential to affect Nix users.

**Pull request closed without merging**